pkg/synchronization/controller.go

```diff
@@ -456,8 +455,31 @@ func (c *Controller) createOrUpdateServiceMetadata(logger *zap.Logger, ns *core_
 	return false, nil
 }
 
-func (c *Controller) getServiceData(user auth.OptionalUser, name voyager.ServiceName) (*creator_v1.Service, error) {
-	return c.ServiceCentral.GetService(context.Background(), user, servicecentral.ServiceName(name))
+func (c *Controller) getCachedServiceData(name voyager.ServiceName) (*creator_v1.Service, bool /* ok */) {
+	c.ServiceCacheMutex.RLock()
+	defer c.ServiceCacheMutex.RUnlock()
+
+	service, ok := c.ServiceCache[name]
+	return service, ok
+}
+
+func (c *Controller) fetchAndCacheServiceData(name voyager.ServiceName) (*creator_v1.Service, error) {
+	c.ServiceCacheMutex.Lock()
```

Comment:
If I'm reading the code correctly this means that all the workers are serialized on this mutex and there is no concurrency here. We don't want that.

Reply:
I can move it after the call to SC, then it will be less impactful.

Reply:
to clarify: given that a single call to SC is the slowest part of this function, moving a lock after it means that the SC requests will be still parallelized, which is the main reason for executing this function from multiple go routines.